arm/proofs/simulator_iclasses.ml

```diff
@@ -459,6 +462,24 @@ let check_insns () =
     (*** st1 (1 register, Post-immediate offset) ***)
     "0x001100100111110111xxxxxxxxxxxx";
 
+    (*** ld1 (1 register, Post-register offset) ***)
+    "0x001100110xxxxx0111xxxxxxxxxxxx";
+
+    (*** st1 (1 register, Post-register offset) ***)
+    "0x001100100xxxxx0111xxxxxxxxxxxx";
+
+    (*** ld1 (1 register, no Post-immediate offset) ***)
+    "0x001100010000000111xxxxxxxxxxxx";
+
+    (*** st1 (1 register, no Post-immediate offset) ***)
+    "0x001100100111110111xxxxxxxxxxxx";
```

Comment:
Does "no Post-immediate offset" mean "no offset"? I checked the manual and this pattern seems different: https://developer.arm.com/documentation/ddi0602/2023-12/SIMD-FP-Instructions/ST1--multiple-structures---Store-multiple-single-element-structures-from-one--two--three--or-four-registers-?lang=en

Reply:
Thanks, Yan.
- yes, "no Post-immediate offset" is equivalent to "no offset" in the documentation. I will change the comments here and in decode.ml
- you're right, I may have copy-pasted without changing. The bit pattern should be the same as the ld1 above it except for the load (L) bit being set to 0. I'll fix it.

I have a question, should these now be moved out of the `skipping_iclasses`?

Reply:
I think they should be kept in there since it belongs to a check for eliminating testing of LD/ST instructions from the assembly.

Reply:
Thank you both for the last checks and improvements! I agree with keeping things in `skipping_iclasses` for now, until we finally have a comprehensive check correlating our cosimulation of memory instructions with the patterns. We have an open issue for this so that we can't forget: https://github.com/awslabs/s2n-bignum/issues/199

Reply:
The question of having separate `LD1`/`ST1` `arm_xxx` instructions is an interesting one, which we might want to consider in the future (and we would definitely need to if we ever model big-endian ARM). If nothing else it could simply be an abbreviation to maintain a better correspondence with the usual assembler syntax, rather as `No_Offset` is defined as a standalone concept but equal semantically to an immediate zero offset.